go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -66,21 +66,29 @@ func NewController(client clientset.Interface,
 
 	// services
 	klog.Info("Setting up event handlers for services")
-	serviceInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
+	_, err := serviceInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
 		AddFunc:    c.onServiceAdd,
 		UpdateFunc: c.onServiceUpdate,
 		DeleteFunc: c.onServiceDelete,
 	}))
+	if err != nil {
+		klog.Warningf("Could not add Event Handler, %v", err)
```

Comment:
why are you not returning the error here?

Reply:
Added error return to services controller newController() so that we can return the error here